repos/system_upgrade/el7toel8/actors/checkgrubcore/actor.py

```diff
@@ -0,0 +1,47 @@
+from leapp.actors import Actor
+from leapp.models import FirmwareFacts, GrubDevice, UpdateGrub
+from leapp.reporting import Report, create_report
+from leapp import reporting
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+
+GRUB_SUMMARY = ('On legacy (BIOS) systems, GRUB core (located in the gap between the MBR and the '
+                'first partition) does not get automatically updated when GRUB is upgraded.')
+
+
+class CheckGrubCore(Actor):
+    """
+    Check whether we are on legacy (BIOS) system and instruct Leapp to upgrade GRUB core
+    """
+
+    name = 'check_grub_core'
+    consumes = (FirmwareFacts, GrubDevice)
+    produces = (Report, UpdateGrub)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        for fact in self.consume(FirmwareFacts):
+            if fact.firmware == 'bios':
+                for dev in self.consume(GrubDevice):
+                    create_report([
+                        reporting.Title(
+                            'GRUB core on {} will be updated during upgrade'.format(dev.grub_device)
+                        ),
+                        reporting.Summary(GRUB_SUMMARY),
+                        reporting.Severity(reporting.Severity.INFO),
+                        reporting.Tags([reporting.Tags.BOOT]),
+                    ])
+                    self.produce(UpdateGrub(grub_device=dev.grub_device))
+                    return
```

Comment:
Looking at that return - so we are processing the first GrubDevice and skipping the rest?

Reply:
we already discussed this offline, this will be improved...